src/mixer/samplerbank.cpp

```diff
@@ -31,30 +37,28 @@ void SamplerBank::slotSaveSamplerBank(double v) {
         return;
     }
 
-    QString fileFilter = tr("Mixxx Sampler Banks (*.xml)");
-    QString samplerBankPath = QFileDialog::getSaveFileName(
-            NULL, tr("Save Sampler Bank"),
+    const QString xmlSuffix = QStringLiteral(".xml");
+    const QString fileFilter = tr("Mixxx Sampler Banks (*%1)").arg(xmlSuffix);
+    QString samplerBankPath = QFileDialog::getSaveFileName(nullptr,
+            tr("Save Sampler Bank"),
             QString(),
-            tr("Mixxx Sampler Banks (*.xml)"),
+            fileFilter,
             &fileFilter);
```

Comment:
looksl like we can't make `fileFilter` const.

```
src/mixer/samplerbank.cpp: In member function ‘void SamplerBank::slotSaveSamplerBank(double)’:
src/mixer/samplerbank.cpp:46:24: error: invalid conversion from ‘const QString*’ to ‘QString*’ [-fpermissive]
             &fileFilter);
                        ^
In file included from /usr/include/qt5/QtWidgets/QFileDialog:1:0,
                 from src/mixer/samplerbank.cpp:3:
/usr/include/qt5/QtWidgets/qfiledialog.h:223:20: error:   initializing argument 5 of ‘static QString QFileDialog::getSaveFileName(QWidget*, const QString&, const QString&, const QString&, QString*, QFileDialog::Options)’ [-fpermissive]
     static QString getSaveFileName(QWidget *parent = 0,
                    ^
```

Reply:
okay.
btw which flags do I have to set in order to see those warnings?

Reply:
Nine, the warning should always happen with the default cmake build. But you should set `-DWARNINGS_FATAL=ON` to treat warnings as errors (that way you can't can't miss them while the build log scrolls by)

Reply:
so I simply remove `const` before the _filter_ declaration?
(sorry can build since scons "doesn't work anymore" here)

Reply:
```
git checkout -b tmp upstream/2.3
git pull origin <branch>
```

Then you can use cmake for building. Make some changes, `git stash`, switch back to the original branch, `git stash pop` and`git branch - D tmp`.

Reply:
And yes, just remove the `const`.

Reply:
> ```
> git checkout -b tmp upstream/2.3
> git pull origin sampler-export
> ```
> 
> Then you can use cmake for building. Make some changes, `git stash`, switch back to the original branch, `git stash pop` and`git branch -D tmp`.

nice trick, thanks!